### PR TITLE
[SPARK-47880][SQL][DOCS] Oracle: Document Mapping Spark SQL Data Types to Oracle

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Documents Mapping Spark SQL Data Types to Oracle

### Why are the changes needed?

documentation improvement

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?

doc build
image

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: docs/sql-data-sources-jdbc.md [hunk 1]
```diff
@@ -1335,3 +1335,109 @@ as the activated JDBC Driver.
...
+      <td></td>
+    </tr>
+    <tr>
+      <td>CharType(n)</td>
+      <td>CHAR(n)</td>
+      <td></td>
+    </tr>
+    <tr>
+      <td>VarcharType(n)</td>
+      <td>VARCHAR2(n)</td>
```
Comment (yaooqinn) [inline docs/sql-data-sources-jdbc.md:1426]: It's VARCHAR2 after #46091 
It does not make any difference to the users, so I didn't add any remarks as less is more.

File: docs/sql-data-sources-jdbc.md [hunk 2]
```diff
@@ -1335,3 +1335,109 @@ as the activated JDBC Driver.
...
+      <th><b>Spark SQL Data Type</b></th>
+      <th><b>Oracle Data Type</b></th>
+      <th><b>Remarks</b></th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>BooleanType</td>
+      <td>NUMBER(1, 0)</td>
+      <td>BooleanType maps to NUMBER(1, 0) as BOOLEAN is introduced since Oracle Release 23c</td>
```
Comment (yaooqinn) [inline docs/sql-data-sources-jdbc.md:1357]: we might introduce a configuration for users to map BOOLEAN instead.

Comment (dongjoon-hyun) [reply docs/sql-data-sources-jdbc.md:1357]: Is it because `Oracle Database 23c` is introduced on April 2023?

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM for AS-IS PR.

For `BOOLEAN`, we can revise it later when we are going to add a new conf.

File: docs/sql-data-sources-jdbc.md (hunk 2 above)
Comment (yaooqinn) [reply docs/sql-data-sources-jdbc.md:1357]: Yes, previous versions didn't have BOOLEAN

**Comment by yaooqinn** [comment]
Merged to master, Thank you @dongjoon-hyun

**Pull request closed without merging**